### PR TITLE
Add ClientCallbackHandler to pass back ApplicationId and TaskUrl

### PR DESCRIPTION
**Pull request by oliverhu**
We current expose taskUrlSet to NotebookSubmitter through exposing that field in TonyClient, that doesn't look good. Also, we need applicationId passed back to `TonyRuntime` .

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,7 +72,7 @@ ext.deps = [
 
 allprojects {
   group = "com.linkedin.tony"
-  project.version = "0.2.0"
```
Comment (erwa) [inline build.gradle]: Why append `-SNAPSHOT`?

File: tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+package com.linkedin.tony;
+
+import com.linkedin.tony.rpc.TaskUrl;
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+
+import java.util.Set;
+
+public interface ClientCallbackHandler {
+    public void onApplicationIdReceived(ApplicationId appId);
+    public void onTaskUrlReceived(Set<TaskUrl> taskUrlSet);
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java]: TaskUrl -> TaskUrls (with an 's' at the end)

File: tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java]: 2019

File: tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import com.linkedin.tony.rpc.TaskUrl;
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+
+import java.util.Set;
+
+public interface ClientCallbackHandler {
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/ClientCallbackHandler.java]: Can you add some Javadoc here and to the interface methods?

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 5]
```diff
@@ -732,6 +738,9 @@ private boolean monitorApplication()
         taskUrls = amRpcClient.getTaskUrls();
         if (!taskUrls.isEmpty()) {
           // Print TaskUrls
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: Move this comment down to before line 744.

File: build.gradle (hunk 1 above)
Comment (oliverhu) [reply build.gradle]: I will revert that. This was for publishing to local maven repo to prevent conflicts with maven central's version.

**Pull request merged**